### PR TITLE
Refactor admin REST API calls into single file

### PR DESCRIPTION
**Pull request by hotzenklotz**
I refactored the majority of all REST API requests from the admin section into a single file. All URLs and `Request.<method>` calls for retrieving, creating, updating and deleting should be gathered here for easy access / testing.

Additionally, requests that contain 1000 results will show a toast to let the users know that they hit the server's delivery limit. See https://discuss.webknossos.org/t/some-lists-cut-data-at-certain-point/262

Future admin views / reactorings should use this as well.

Tests will be added after the snapshot testing PR #2034 has been merged.

### Steps to test:
- Test all the admin views: Teams, Projects, Users, TaskTypes, Scripts

@philippotto @daniel-wer Comments? Ideas? Suggestions?

------
- [x] Ready for review

### CODE REVIEW

**Comment by philippotto** [comment]
This refactoring looks really good!

> Additionally, requests that contain 1000 results will show a toast to let the users know that they hit the server's delivery limit. See https://discuss.webknossos.org/t/some-lists-cut-data-at-certain-point/262

I feel like the server should take care of this, by sending a message along the response (which will be automatically toasted anyway). If there happen to be exactly 1000 items by coincidence, there will be a warning even though it's not really justified. Somehow, this feels hacky.

> Tests will be added after the snapshot testing PR #2034 has been merged.

Yeah, this should be a good opportunity to test the tests!

**Comment by daniel-wer** [comment]
Very nice :unicorn:

**Comment by normanrz** [comment]
I think using 1000 as heuristic for the warning is fine enough. Our API isn't paginated anyway, so there are no means to retrieve more than 1000 docs.

File: app/assets/javascripts/messages.js [hunk 1]
```diff
@@ -33,4 +33,6 @@ In order to restore the current window, a reload is necessary.`,
   "script.delete": "Do you really want to delete this script?",
   "team.delete": "Do you really want to delete this team?",
   "taskType.delete": "Do you really want to delete this task type?",
+  "request.max_item_count_alert":
+    "Your request returned more than 1000 results. More results might be available on the server but were omitted for technical reseaons.",
```
Comment (philippotto) [inline app/assets/javascripts/messages.js]: "reseaons" should probably be "reasons".

**Pull request merged**